### ISSUE TITLE
h5py=3.3.0 causes error 

### ISSUE THREAD

**Issue by romainGuiet** [stardist/stardist#150]
**Describe the bug**
In recent install (build yesterday) h5py 3.3.0 was installed and got error with model command like  ```model.train() ``` , or when loading the model.

**Environment (please complete the following information):**
 - StarDist version  0.6.2
 - CSBDeep version  0.6.2
 - TensorFlow version 1.15.0 (env made to be able to use model in Fiji directly)
 - OS: Linux 
 - GPU memory (if applicable): 11 GB]

You may run this code and paste the output:
 ``` model = StarDist2D(None, name=user_model, basedir='../models')``` which returns error : 
``` ---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-8-c21cf09e7bd8> in <module>
----> 1 model = StarDist2D(None, name=user_model, basedir='../models')
      2 #X, Y = helpers.load_all_data( data_folder, is_norm_independently )
      3 # Optimize the thresholds only on the user's data, regardless of whether it was used for training or validation
      4 model.optimize_thresholds( X_val_usr, Y_val_usr )

/opt/conda/envs/py37/lib/python3.7/site-packages/stardist/models/model2d.py in __init__(self, config, name, basedir)
    253     def __init__(self, config=Config2D(), name=None, basedir='.'):
    254         """See class docstring."""
--> 255         super().__init__(config, name=name, basedir=basedir)
    256 
    257 

/opt/conda/envs/py37/lib/python3.7/site-packages/stardist/models/base.py in __init__(self, config, name, basedir)
    157 
    158     def __init__(self, config, name=None, basedir='.'):
--> 159         super().__init__(config=config, name=name, basedir=basedir)
    160         threshs = dict(prob=None, nms=None)
    161         if basedir is not None:

/opt/conda/envs/py37/lib/python3.7/site-packages/csbdeep/models/base_model.py in __init__(self, config, name, basedir)
    111         self.keras_model = self._build()
    112         if config is None:
--> 113             self._find_and_load_weights()
    114 
    115 

/opt/conda/envs/py37/lib/python3.7/site-packages/csbdeep/models/base_model.py in wrapper(*args, **kwargs)
     30                 warn is False or warnings.warn("Suppressing call of '%s' (due to basedir=None)." % f.__name__)
     31             else:
---> 32                 return f(*args, **kwargs)
     33         return wrapper
     34     return _suppress_without_basedir

/opt/conda/envs/py37/lib/python3.7/site-packages/csbdeep/models/base_model.py in _find_and_load_weights(self, prefer)
    165         weights_chosen = weights_preferred[0] if len(weights_preferred)>0 else weights_files[0]
    166         print("Loading network weights from '%s'." % weights_chosen.name)
--> 167         self.load_weights(weights_chosen.name)
    168 
    169 

/opt/conda/envs/py37/lib/python3.7/site-packages/csbdeep/models/base_model.py in wrapper(*args, **kwargs)
     30                 warn is False or warnings.warn("Suppressing call of '%s' (due to basedir=None)." % f.__name__)
     31             else:
---> 32                 return f(*args, **kwargs)
     33         return wrapper
     34     return _suppress_without_basedir

/opt/conda/envs/py37/lib/python3.7/site-packages/csbdeep/models/base_model.py in load_weights(self, name)
    182             Name of HDF5 weight file (as saved during or after training).
    183         """
--> 184         self.keras_model.load_weights(str(self.logdir/name))
    185 
    186 

/opt/conda/envs/py37/lib/python3.7/site-packages/keras/engine/saving.py in load_wrapper(*args, **kwargs)
    490                 os.remove(tmp_filepath)
    491             return res
--> 492         return load_function(*args, **kwargs)
    493 
    494     return load_wrapper

/opt/conda/envs/py37/lib/python3.7/site-packages/keras/engine/network.py in load_weights(self, filepath, by_name, skip_mismatch, reshape)
   1228             else:
   1229                 saving.load_weights_from_hdf5_group(
-> 1230                     f, self.layers, reshape=reshape)
   1231             if hasattr(f, 'close'):
   1232                 f.close()

/opt/conda/envs/py37/lib/python3.7/site-packages/keras/engine/saving.py in load_weights_from_hdf5_group(f, layers, reshape)
   1181     """
   1182     if 'keras_version' in f.attrs:
-> 1183         original_keras_version = f.attrs['keras_version'].decode('utf8')
   1184     else:
   1185         original_keras_version = '1'

AttributeError: 'str' object has no attribute 'decode'
```

Cheers

**Comment by romainGuiet** [comment]
Downgrading to h5py 2.10.0 solve the issue

**Comment by maweigert** [comment]
Hi @romainGuiet,

Yes, there seemed to be a [keras issue](https://github.com/tensorflow/tensorflow/issues/44467) with older tensorflow versions, so downgrading as you did is indeed the best solution.

Thanks for reporting!

**Comment by uschmidt83** [comment]
Actually, the [csbdeep releases](https://github.com/CSBDeep/CSBDeep/releases) 0.6.1 and 0.6.2 specifically addressed this issue, enforcing `h5py < 3.0.0` (except for Python 3.9, but you're not using this since TF 1.x doesn't support it).

My best guess is that you installed some other package after csbdeep/stardist, which required a newer version of h5py.

**Comment by romainGuiet** [comment]
Hi @maweigert , Hi @uschmidt83 ,

>My best guess is that you installed some other package after csbdeep/stardist, which required a newer version of h5py.

That could definitely be the reason ! it's within a RENKU project with some conda and pip installs so one of the other packages might indeed mess things up!!!

I was thinking that posting it here would make the solution easier to find for someone with a similar issue!

Cheers,

Romain
